src/containers/Storage/Storage.tsx

```diff
@@ -230,36 +170,29 @@ export const Storage = ({database, nodeId, groupId, pDiskId}: StorageProps) => {
     };
 
     const renderControls = () => {
-        const entitiesCountCurrent = isGroups ? storageGroups.length : storageNodes.length;
-
-        const columnsToSelect = isGroups
-            ? storageGroupsColumnsToSelect
-            : storageNodesColumnsToSelect;
-
-        const handleSelectedColumnsUpdate = isGroups
-            ? setStorageGroupsSelectedColumns
-            : setStorageNodesSelectedColumns;
-
         return (
-            <StorageControls
-                searchValue={filter}
-                handleSearchValueChange={handleTextFilterChange}
-                withTypeSelector
-                storageType={storageType}
-                handleStorageTypeChange={handleStorageTypeChange}
-                visibleEntities={visibleEntities}
-                handleVisibleEntitiesChange={handleGroupVisibilityChange}
-                nodesUptimeFilter={uptimeFilter}
-                handleNodesUptimeFilterChange={handleUptimeFilterChange}
-                groupsUsageFilter={usageFilter}
-                groupsUsageFilterOptions={usageFilterOptions}
-                handleGroupsUsageFilterChange={handleUsageFilterChange}
-                entitiesCountCurrent={entitiesCountCurrent}
-                entitiesCountTotal={entitiesCount.total}
-                entitiesLoading={isLoading}
-                columnsToSelect={columnsToSelect}
-                handleSelectedColumnsUpdate={handleSelectedColumnsUpdate}
-            />
+            <React.Fragment>
+                {isGroups ? (
+                    <StorageGroupsControls
+                        withTypeSelector
+                        entitiesCountCurrent={storageGroups.length}
+                        entitiesCountTotal={groupsTotalCount}
+                        entitiesLoading={isLoading}
+                        columnsToSelect={storageGroupsColumnsToSelect}
+                        handleSelectedColumnsUpdate={setStorageGroupsSelectedColumns}
+                    />
+                ) : null}
+                {isNodes ? (
+                    <StorageNodesControls
+                        withTypeSelector
+                        entitiesCountCurrent={storageNodes.length}
+                        entitiesCountTotal={nodesTotalCount}
+                        entitiesLoading={isLoading}
+                        columnsToSelect={storageNodesColumnsToSelect}
+                        handleSelectedColumnsUpdate={setStorageNodesSelectedColumns}
+                    />
+                ) : null}
```

Comment:
Added different controls to the same component. As the next step `Storage` could be split into two components like `PaginatedStorage`